CYD-Klipper/src/core/device/ESP32-CROWPANEL_28R.cpp

```diff
@@ -0,0 +1,111 @@
+#ifdef CYD_SCREEN_DRIVER_ESP32_CROWPANEL_28R
+#include "../screen_driver.h"
+
+#ifdef CYD_SCREEN_VERTICAL
+    #error "Vertical screen not supported with the ESP32_CROWPANEL_28R driver"
+#endif
+
+#include <SPI.h>
+#include <TFT_eSPI.h>
+#include "../../conf/global_config.h"
+#include "lvgl.h"
+#include <TFT_eSPI.h>
+#include "../lv_setup.h"
+
+#define CPU_FREQ_HIGH 240
+#define CPU_FREQ_LOW 80
+#define TOUCH_THRESHOLD 600
+
+static lv_disp_draw_buf_t draw_buf;
+static lv_color_t buf[CYD_SCREEN_HEIGHT_PX * CYD_SCREEN_WIDTH_PX / 10];
+
+TFT_eSPI tft = TFT_eSPI();
+
+uint16_t touchX, touchY;
+
+void screen_setBrightness(byte brightness)
+{
+    // calculate duty, 4095 from 2 ^ 12 - 1
+    uint32_t duty = (4095 / 255) * brightness;
+
+    // write duty to LEDC
+    ledcWrite(0, duty);
+}
+
+void screen_lv_flush(lv_disp_drv_t *disp, const lv_area_t *area, lv_color_t *color_p)
+{
+    uint32_t w = (area->x2 - area->x1 + 1);
+    uint32_t h = (area->y2 - area->y1 + 1);
+
+    tft.startWrite();
+    tft.setAddrWindow(area->x1, area->y1, w, h);
+    tft.pushColors((uint16_t *)&color_p->full, w * h, true);
+    tft.endWrite();
+
+    lv_disp_flush_ready(disp);
+}
+
+void screen_lv_touchRead(lv_indev_drv_t *indev_driver, lv_indev_data_t *data)
+{
+    if (tft.getTouch( &touchX, &touchY, TOUCH_THRESHOLD))
+    {
+        data->state = LV_INDEV_STATE_PR;
+        data->point.x = touchX;
+        data->point.y = touchY;
+    }
+    else
+    {
+        data->state = LV_INDEV_STATE_REL;
+    }
+}
+
+void set_invert_display(){
+    tft.invertDisplay(get_current_printer_config()->invert_colors);
+}
+
+void screen_setup()
+{
+    uint16_t calData[5] = { 189, 3416, 359, 3439, 1};
+
+    lv_init();
+
+    tft.init();
+    tft.fillScreen(TFT_BLACK);
+    tft.invertDisplay(false);
+    delay(300);
+
+    tft.setRotation(1);
+    tft.setTouch( calData );
+
+    if (global_config.display_mode) {
```

Comment:
Is this also needed on this display?

Reply:
In the original CrowPanel firmware I can see this was added, and this changes only made the display work.

Reply:
@MadRajib This is a setting. If it's required to make the display work, it shouldn't be conditional.

This option is hidden on any non-2.8"R CYD screen as well.

Reply:
Sorry, I though you were asking about the _tft.setTouch( calData );_

Agreed this shouldn't be used for this display type, I have removed it on the latest patch.